lxd/storage/backend_lxd.go

```diff
@@ -2769,6 +2769,55 @@ func (b *lxdBackend) DeleteImage(fingerprint string, op *operations.Operation) e
 	return nil
 }
 
+// DeleteImageTx removes an image from the database and underlying storage device if needed.
+// TODO: Replace DeleteImage with this method.
+func (b *lxdBackend) DeleteImageTx(fingerprint string, tx *db.ClusterTx, op *operations.Operation) error {
```

Comment:
@masnax as deleting volumes from a storage pool can be slow I would definitely not be confident that running multiple of these operations inside a single transaction would be a good idea, as it would likely hold up the whole database from having writes performed by other queries whilst this is in effect (sqlite is write locked at the db level).